kernel/sched.c

```diff
@@ -1619,3 +1619,43 @@ static inline void z_vrfy_k_thread_abort(k_tid_t thread)
 }
 #include <syscalls/k_thread_abort_mrsh.c>
 #endif /* CONFIG_USERSPACE */
+
+/*
+ * future scheduler.h API implementations
+ */
+bool z_sched_callback_wake(_wait_q_t *wait_q, int swap_retval, void *swap_data,
+			   z_sched_wake_cb_t cb, void *obj, void *context)
+{
+	struct k_thread *thread;
+	bool ret = false;
+
+	LOCKED(&sched_spinlock) {
+		thread = _priq_wait_best(&wait_q->waitq);
+
+		if (thread != NULL) {
+			if (cb != NULL) {
+				cb(thread, obj, context);
```

Comment:
Not loving the idea of having an API, even an internal one, provide a callback that gets called with an internal lock held.  Note that if that callback then tries to do anything at all to do with the scheduler, it will cause a deadlock.